lib/components/term.js

```diff
@@ -98,9 +106,24 @@ export default class Term extends Component {
 
     this.getScreenNode().addEventListener('mouseup', this.handleMouseUp);
     this.getScreenNode().addEventListener('mousedown', this.handleMouseDown);
+    this.getScreenNode().addEventListener('contextmenu', this.handleRightClick);
     terms[this.props.uid] = this;
   }
 
+  handleRightClick(e) {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const {quickEdit} = config.getConfig();
+    const show = !quickEdit;
+
+    const contextMenu = Menu.buildFromTemplate(contextMenuTemplate);
+    if (show) {
+      // throttle popup so right click has time to highlight text
+      setTimeout(() => contextMenu.popup(getCurrentWindow()), 100);
```

Comment:
I am really not fan to call this here.
You are right, context menu should be contextual to right-clicked term.
But I would prefer to dispatch a redux action (and maybe reduce it to retain in state which term is concerned), make a rpc call and popup context menu from app.

`Term` is a component and should not have any knowledge of `electron`.